library/ssl_msg.c

```diff
@@ -5670,7 +5672,9 @@ static int ssl_write_real(mbedtls_ssl_context *ssl,
          */
         ssl->out_msglen  = len;
         ssl->out_msgtype = MBEDTLS_SSL_MSG_APPLICATION_DATA;
-        memcpy(ssl->out_msg, buf, len);
+        if (len > 0) {
+            memcpy(ssl->out_msg, buf, len);
+        }
 
         if ((ret = mbedtls_ssl_write_record(ssl, SSL_FORCE_FLUSH)) != 0) {
```

Comment:
As above, is it possible to have `ssl->out_msglen == 0` and `ret = mbedtls_ssl_write_record(ssl, SSL_FORCE_FLUSH)) != 0` ?

Reply:
I'm not able to trigger that case, so maybe not? But I don't think the behaviour of `mbedtls_ssl_write_record` should be affected by this change.